yorkie/auth/webhook.go

```diff
@@ -49,29 +54,70 @@ func VerifyAccess(ctx context.Context, be *backend.Backend, info *types.AccessIn
 		return err
 	}
 
-	// TODO(hackerwins): We need to apply retryBackoff in case of failure
-	resp, err := http.Post(
-		be.Config.AuthorizationWebhookURL,
-		"application/json",
-		bytes.NewBuffer(reqBody),
-	)
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if err := resp.Body.Close(); err != nil {
-			log.Logger.Error(err)
+	return withExponentialBackoff(be.Config, func() (int, error) {
+		resp, err := http.Post(
+			be.Config.AuthorizationWebhookURL,
+			"application/json",
+			bytes.NewBuffer(reqBody),
+		)
+		if err != nil {
+			return 0, err
 		}
-	}()
 
-	authResp, err := types.NewAuthWebhookResponse(resp.Body)
-	if err != nil {
+		defer func() {
+			if err := resp.Body.Close(); err != nil {
+				log.Logger.Error(err)
+			}
+		}()
+
+		if http.StatusOK != resp.StatusCode {
+			return resp.StatusCode, fmt.Errorf("%d: %w", resp.StatusCode, ErrUnexpectedStatusCode)
+		}
+
+		authResp, err := types.NewAuthWebhookResponse(resp.Body)
+		if err != nil {
+			return resp.StatusCode, err
+		}
+
+		if !authResp.Allowed {
+			return resp.StatusCode, fmt.Errorf("%s: %w", authResp.Reason, ErrNotAllowed)
+		}
+
+		return resp.StatusCode, nil
+	})
+}
+
+func withExponentialBackoff(config *backend.Config, webhookFn func() (int, error)) error {
+	statusCode, err := webhookFn()
+	if !shouldRetry(statusCode) {
 		return err
 	}
 
-	if !authResp.Allowed {
-		return fmt.Errorf("%s: %w", authResp.Reason, ErrNotAllowed)
+	var retries uint64
+	for {
+		if retries == config.AuthorizationWebhookMaxRetries {
+			return err
+		}
+
+		time.Sleep(waitInterval(retries, config.AuthorizationWebhookMaxWaitInterval))
+
+		statusCode, err := webhookFn()
+		if !shouldRetry(statusCode) {
+			return err
+		}
+		retries++
 	}
+}
+
+func waitInterval(retries uint64, maxWaitInterval uint64) time.Duration {
```

Comment:
It would be nice to express the unit of interval.

```suggestion
func waitInterval(retries uint64, maxWaitInterval time.Duration) time.Duration {
```